src/live-reload-middleware.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Request, RequestHandler, Response} from 'express';
+import * as dom5 from 'dom5';
+import * as parse5 from 'parse5';
+
+import {transformResponse, getContentType, isSuccessful} from './transform-middleware';
+
+export const liveReloadAppend: RequestHandler =
+  transformResponse({
+    shouldTransform(_request: Request, response: Response) {
+      return isSuccessful(response) && getContentType(response) === 'text/html';
+    },
+
+    transform(_request: Request, _response: Response, body: string): string {
```

Comment:
This transformation is required to not generate invalid HTML. Appending the import simply to body resulted in invalid HTML which the browser did not want to parse correctly.